pkgs/top-level/all-packages.nix

```diff
@@ -17908,6 +17908,22 @@ in
     ];
   };
 
+  # Needs a `date` and a `sha256` argument. If version differs from `linux_testing` it also needs
+  # `extraMeta.branch` and `extraMeta.rc`.
+  linux_next = args: with lib; callPackage ../os-specific/linux/kernel/linux-next.nix (rec {
+    extraMeta.branch = args.extraMeta.branch or linux_testing.meta.branch;
+    extraMeta.rc = args.extraMeta.rc or (last (splitString "-" linux_testing.version));
+    kernelPatches = args.kernelPatches or linux_testing.kernelPatches;
+    sha256 = args.sha256 or lib.fakeSha256;
+    version = args.version or (
+      let branch = extraMeta.branch + ".0";
+          rc = extraMeta.rc;
+      in with lib; concatStringsSep "-" (
```

Comment:
`with lib;` is superfluous here, since it's already there in the top level of the function (`args: with lib;`)